ra-optimisation/search-tree-rao/src/main/java/com/farao_community/farao/search_tree_rao/process/search_tree/Tree.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (c) 2019, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ *  License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.farao_community.farao.search_tree_rao.process.search_tree;
+
+import com.farao_community.farao.commons.FaraoException;
+import com.farao_community.farao.data.crac_api.Crac;
+import com.farao_community.farao.data.crac_api.NetworkAction;
+import com.farao_community.farao.data.crac_api.UsageMethod;
+import com.farao_community.farao.ra_optimisation.RaoComputationResult;
+import com.farao_community.farao.rao_api.RaoParameters;
+import com.powsybl.iidm.network.Network;
+
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * The "tree" is one of the core object of the search-tree algorithm.
+ * It aims at finding a good combination of Network Actions.
+ *
+ * The tree is composed of leaves which evaluate the impact of Network Actions,
+ * one by one. The tree is orchestrating the leaves : it looks for a smart
+ * routing among the leaves in order to converge as quick as possible to a local
+ * minimum of the objective function.
+ *
+ * @author Joris Mancini {@literal <joris.mancini at rte-international.com>}
+ * @author Baptiste Seguinot {@literal <baptiste.seguinot at rte-international.com>}
```

Comment:
We have job position open for experienced developers.

Reply:
changed as well